### PR TITLE
feat: add sign in with GitHub oauth

### PR DESCRIPTION
**Pull request by isavega**
# Base PR
- PR based on main

# Context
- With this PR, it is possible to login with github with omniauth gem

# Changelog
- Add new gem: omniauth-github
- Add a new oauth2 provider (github)
- Add github sign in button to devise links view
- Add handle_auth method to `OmniauthCallbacksController` to handle multiple providers 
- Add new secrets to devise configuration 

# QA
- Create .env file with the credentials
- please check if you have a public email in your github account. Otherwise, it is not possible to use this registration method
- Run: docker compose build
- Run: docker compose run web rails db:reset
- Run: docker compose run web rails db:migrate
- [FRONT] On the main branch go to http://localhost:3001/users/sign_in

### CODE REVIEW

**Review by vpulgar — APPROVED** [review]

File: app/views/devise/shared/_links.html.erb [hunk 1]
```diff
@@ -3,16 +3,24 @@
 <% end %>
 
 <%- if devise_mapping.omniauthable? %>
-  <%- resource_class.omniauth_providers.each do |provider| %>
+ 
     <%= form_for "Login with Google",
-        url: omniauth_authorize_path(resource_name, provider), 
+        url: omniauth_authorize_path(resource_name, resource_class.omniauth_providers[0]),
```
Comment (AlehandroL) [inline app/views/devise/shared/_links.html.erb:8]: This depends on the order of the array. Can't you call it directly by name?

File: app/views/devise/shared/_links.html.erb [hunk 2]
```diff
@@ -3,16 +3,24 @@
...
         method: :post,
-        class:"form",
+        class:"google-form",
         data: {turbo: "false"} do |f| %>
           <%= f.submit type: "image", src: url_for("/assets/btn_google_dark_normal_ios.svg"), class:"google-icon"%>
           <%= f.submit "Sign in with Google", class:"google-text"%>
     <% end %>
-  <% end %>
+        <%= form_for "Login with Github",
+        url: omniauth_authorize_path(resource_name, resource_class.omniauth_providers[1]),
```
Comment (AlehandroL) [inline app/views/devise/shared/_links.html.erb:16]: Same as above

**Review by AlehandroL — CHANGES_REQUESTED** [review]
Looks great, only a minor change is requested.

File: app/views/devise/shared/_links.html.erb (hunk 1 above)
Comment (isavega) [reply app/views/devise/shared/_links.html.erb:8]: Good catch

**Review by jjsantos01 — CHANGES_REQUESTED** [review]

File: app/views/devise/shared/_links.html.erb [hunk 3]
```diff
@@ -3,16 +3,24 @@
...
           <%= f.submit "Sign in with Google", class:"google-text"%>
     <% end %>
-  <% end %>
+        <%= form_for "Login with Github",
+        url: omniauth_authorize_path(resource_name, resource_class.omniauth_providers[1]), 
+        method: :post,
+        class:"github-form",
+        data: {turbo: "false"} do |f| %>
+          <%= f.submit type: "image", src: url_for("/assets/GitHub-Mark-120px-plus.png"), class:"github-icon"%>
+          <%= f.submit "Login with Github", class:"github-text"%>
```
[the remark this answers is not in the document]
Comment (isavega) [reply app/views/devise/shared/_links.html.erb:21]: Ok!

[the remark this answers is not in the document]
Comment (isavega) [reply app/views/devise/shared/_links.html.erb:21]: Ok

**Review by jjsantos01 — APPROVED** [review]
Great PR! Tested on local and everything worked fine

**Review by AlehandroL — APPROVED** [review]

**Pull request merged**